apps/vaults/hooks/useSolverCowswap.ts

```diff
@@ -276,22 +302,35 @@ export function useSolverCowswap(): TSolverContext {
 			return (toNormalizedBN(0));
 		}
 		const quoteResult = await getQuote(_request, true);
-		return toNormalizedBN(formatBN(quoteResult?.quote?.buyAmount), _request.outputToken.decimals);
+		return (
+			toNormalizedBN(
+				toBigInt(quoteResult?.quote?.buyAmount.toString()),
+				_request.outputToken.decimals || 18
+			)
+		);
 	}, [getQuote, isDisabled]);
 
 	/* 🔵 - Yearn Finance ******************************************************
 	** Retrieve the allowance for the token to be used by the solver. This will
 	** be used to determine if the user should approve the token or not.
 	**************************************************************************/
 	const onRetrieveAllowance = useCallback(async (shouldForceRefetch?: boolean): Promise<TNormalizedBN> => {
-		if (!request?.current || isDisabled || !request?.current?.inputToken?.solveVia?.includes(Solver.COWSWAP)) {
+		if (isDisabled) {
 			return toNormalizedBN(0);
 		}
+		assert(request.current, 'Request is not defined');
+		assert(request?.current?.inputToken?.solveVia?.includes(Solver.COWSWAP), 'Input token is not supported by Cowswap');
 
-		const key = allowanceKey(safeChainID, toAddress(request.current.inputToken.value), toAddress(request.current.outputToken.value), toAddress(request.current.from));
+		const key = allowanceKey(
+			safeChainID,
+			toAddress(request.current.inputToken.value),
+			toAddress(request.current.outputToken.value),
+			toAddress(request.current.from)
+		);
 		if (existingAllowances.current[key] && !shouldForceRefetch) {
 			return existingAllowances.current[key];
 		}
+		assert(provider, 'Provider is not defined');
 		const allowance = await approvedERC20Amount(
```

Comment:
This combo, checking if `provider` is defined before a `approvedERC20Amount` is used in a bunch of places, maybe better to move inside that function to reduce the number of asserts. Wdyt?